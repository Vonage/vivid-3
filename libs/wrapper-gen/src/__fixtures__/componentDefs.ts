import { ComponentDef } from '../metadata/ComponentDef';

export const minimalComponent: ComponentDef = {
	name: 'example-component',
	className: 'ExampleComponent',
	wrappedClassName: 'VExampleComponent',
	vividModulePath:
		'libs/components/src/lib/example-component/example-component.ts',
	registerFunctionName: 'registerExampleComponent',
	attributes: [],
	events: [],
	vueModels: [],
	methods: [],
	slots: [],
	localTypeDefs: {},
};

export const exampleComponent: ComponentDef = {
	name: 'example-component',
	className: 'ExampleComponent',
	wrappedClassName: 'VExampleComponent',
	vividModulePath:
		'libs/components/src/lib/example-component/example-component.ts',
	registerFunctionName: 'registerExampleComponent',
	description: 'This is an example component',
	attributes: [
		{
			name: 'example-attribute',
			description: 'This is an example attribute',
			forwardTo: {
				type: 'attribute',
				name: 'example-attribute',
			},
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'type-union',
			forwardTo: {
				type: 'attribute',
				name: 'type-union',
			},
			type: [
				{ text: 'string', vuePropType: 'String' },
				{ text: 'number', vuePropType: 'Number' },
			],
		},
		{
			name: 'imported-type',
			forwardTo: {
				type: 'property',
				name: 'importedType',
			},
			type: [
				{
					text: 'Imported',
					vuePropType: 'Object',
					importFromModule: 'module',
				},
			],
		},
		{
			name: 'value',
			forwardTo: {
				type: 'attribute',
				name: 'value',
			},
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'start',
			forwardTo: {
				type: 'attribute',
				name: 'start',
			},
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'forced-dom-prop',
			forwardTo: {
				type: 'property',
				name: 'forcedDomProp',
			},
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'boolean-attribute',
			forwardTo: {
				type: 'attribute',
				name: 'boolean-attribute',
				boolean: true,
			},
			type: [
				{
					text: 'boolean',
					vuePropType: 'Boolean',
				},
			],
		},
	],
	events: [
		{
			name: 'example-event',
			description: 'This is an example event',
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'input',
			type: [
				{
					text: 'Event',
					vuePropType: 'Event',
				},
			],
		},
		{
			name: 'input:start',
			type: [
				{
					text: 'CustomEvent<{a: string | number}>',
					vuePropType: 'CustomEvent',
				},
			],
		},
	],
	vueModels: [
		{
			name: 'modelValue',
			attributeName: 'value',
			eventNames: ['input'],
			valueMapping: '(event.target as any).value',
		},
		{
			name: 'sameEvent',
			attributeName: 'example-attribute',
			eventNames: ['input'],
			valueMapping: '(event.target as any).exampleAttribute',
		},
		{
			name: 'start',
			attributeName: 'start',
			eventNames: ['input:start'],
			valueMapping: '(event.target as any).start',
		},
	],
	methods: [
		{
			name: 'exampleMethod',
			description: 'This is an example method',
			args: [
				{
					name: 'exampleArg',
					type: [
						{
							text: 'string',
							vuePropType: 'String',
						},
					],
				},
			],
			returnType: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
	],
	slots: [
		{
			name: 'example-slot',
			description: 'This is an example slot',
		},
		{
			name: 'no-description',
		},
	],
	localTypeDefs: {},
};
