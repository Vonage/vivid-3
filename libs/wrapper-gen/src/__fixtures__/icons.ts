export const icons = ['1-icon-1', 'icon-2'];
