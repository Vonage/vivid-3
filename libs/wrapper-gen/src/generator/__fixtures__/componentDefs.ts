import { ComponentDef } from '../ComponentDef';

export const minimalComponent: ComponentDef = {
	name: 'example-component',
	className: 'ExampleComponent',
	wrappedClassName: 'VExampleComponent',
	vividModulePath:
		'libs/components/src/lib/example-component/example-component.ts',
	registerFunctionName: 'registerExampleComponent',
	attributes: [],
	events: [],
	vueModels: [],
	methods: [],
	slots: [],
	localTypeDefs: {},
};

export const exampleComponent: ComponentDef = {
	name: 'example-component',
	className: 'ExampleComponent',
	wrappedClassName: 'VExampleComponent',
	vividModulePath:
		'libs/components/src/lib/example-component/example-component.ts',
	registerFunctionName: 'registerExampleComponent',
	description: 'This is an example component',
	attributes: [
		{
			name: 'example-attribute',
			description: 'This is an example attribute',
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'type-union',
			type: [
				{ text: 'string', vuePropType: 'String' },
				{ text: 'number', vuePropType: 'Number' },
			],
		},
		{
			name: 'imported-type',
			type: [
				{
					text: 'Imported',
					vuePropType: 'Object',
					importFromModule: 'module',
				},
			],
		},
		{
			name: 'value',
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'start',
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'forced-dom-prop',
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
			forceDomProp: true,
		},
	],
	events: [
		{
			name: 'example-event',
			description: 'This is an example event',
			type: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
		{
			name: 'input',
			type: [
				{
					text: 'Event',
					vuePropType: 'Event',
				},
			],
		},
		{
			name: 'input:start',
			type: [
				{
					text: 'CustomEvent<{a: string | number}>',
					vuePropType: 'CustomEvent',
				},
			],
		},
	],
	vueModels: [
		{
			name: 'modelValue',
			attributeName: 'value',
			eventName: 'input',
			valueMapping: '(event.target as any).value',
		},
		{
			name: 'start',
			attributeName: 'start',
			eventName: 'input:start',
			valueMapping: '(event.target as any).start',
		},
	],
	methods: [
		{
			name: 'exampleMethod',
			description: 'This is an example method',
			args: [
				{
					name: 'exampleArg',
					type: [
						{
							text: 'string',
							vuePropType: 'String',
						},
					],
				},
			],
			returnType: [
				{
					text: 'string',
					vuePropType: 'String',
				},
			],
		},
	],
	slots: [
		{
			name: 'example-slot',
			description: 'This is an example slot',
		},
		{
			name: 'no-description',
		},
	],
	localTypeDefs: {},
};
