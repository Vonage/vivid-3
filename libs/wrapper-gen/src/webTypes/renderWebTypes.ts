import { ComponentDef } from '../metadata/ComponentDef';
import { getTagFromComponentDefinition } from './tags';

export function renderWebTypes(
	componentDefs: ComponentDef[],
	packageVersion: string
): string {
	return JSON.stringify(
		{
			framework: 'vue',
			name: '@vonage/vivid-vue',
			version: packageVersion,
			contributions: {
				html: {
					'description-markup': 'markdown',
					'types-syntax': 'typescript',
					tags: componentDefs.map(getTagFromComponentDefinition),
				},
			},
		},
		null,
		1
	);
}
