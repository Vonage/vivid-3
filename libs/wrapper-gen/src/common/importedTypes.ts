import { fetchIconsManifest, iconTypeFromManifest } from './icons';
import { Metadata } from './metadata';
import { makeTypeResolver } from './types';

export const makeImportedTypesResolver = async (metadata: Metadata) =>
	makeTypeResolver({
		IconId: iconTypeFromManifest(
			await fetchIconsManifest(metadata.iconsManifestUrl)
		),
	});
