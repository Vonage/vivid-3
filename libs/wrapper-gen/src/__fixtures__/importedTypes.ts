import { makeTypeResolver } from '../common/types';
import { iconTypeFromManifest } from '../common/icons';
import { iconsManifestStub } from './icons';

export const importedTypesResolverStub = makeTypeResolver({
	IconId: iconTypeFromManifest(iconsManifestStub),
});
