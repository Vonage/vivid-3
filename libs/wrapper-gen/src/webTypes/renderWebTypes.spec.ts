import { exampleComponent } from '../__fixtures__/componentDefs';
import { renderWebTypes } from './renderWebTypes';
import { importedTypesResolverStub } from '../__fixtures__/importedTypes';

describe('renderWebTypes', () => {
	it('should render web types', () => {
		expect(
			renderWebTypes([exampleComponent], importedTypesResolverStub, '1.2.3')
		).toMatchSnapshot();
	});
});
