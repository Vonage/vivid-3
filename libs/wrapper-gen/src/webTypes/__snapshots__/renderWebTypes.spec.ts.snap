// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renderWebTypes > should render web types 1`] = `
"{
 "framework": "vue",
 "name": "@vonage/vivid-vue",
 "version": "1.2.3",
 "contributions": {
  "html": {
   "description-markup": "markdown",
   "types-syntax": "typescript",
   "tags": [
    {
     "name": "VExampleComponent",
     "description": "This is an example component",
     "attributes": [
      {
       "name": "exampleAttribute",
       "description": "This is an example attribute",
       "value": {
        "kind": "expression",
        "type": "string"
       }
      },
      {
       "name": "typeUnion",
       "value": {
        "kind": "expression",
        "type": "string | number"
       }
      },
      {
       "name": "iconType",
       "value": {
        "kind": "expression",
        "type": "'1-icon-1' | 'icon-2'"
       }
      },
      {
       "name": "value",
       "value": {
        "kind": "expression",
        "type": "string"
       }
      },
      {
       "name": "start",
       "value": {
        "kind": "expression",
        "type": "string"
       }
      },
      {
       "name": "forcedDomProp",
       "value": {
        "kind": "expression",
        "type": "string"
       }
      },
      {
       "name": "booleanAttribute",
       "value": {
        "kind": "expression",
        "type": "boolean"
       }
      }
     ],
     "events": [],
     "slots": [
      {
       "name": "example-slot",
       "description": "This is an example slot"
      },
      {
       "name": "no-description"
      }
     ],
     "source": {
      "module": "@vonage/vivid-vue",
      "symbol": "VExampleComponent"
     },
     "priority": "highest"
    }
   ]
  }
 }
}"
`;
